pages/latest/[urlTitle].tsx

```diff
@@ -41,20 +45,59 @@ const componentsUI: MDXProviderComponentsProp = {
     a: LinkMarkdown
 };
 
-const UrlTitle = ({ data, source, error }: PageData): JSX.Element => {
+const UrlTitle = ({ data, source, toc, error }: PageData): JSX.Element => {
     const content = source ? hydrate(source, { components: componentsUI }) : null;
+    // ---- Table Of Content Active Item
+    const [tocActive, setTocActive] = useState(null);
+    const observer = useRef(null);
+    const options = {
+        root: null,
+        rootMargin: '0px 0px -70% 0px',
+        threshold: 1
+    };
+    useEffect(() => {
+        const targets = document.querySelectorAll('h2,h3');
+        if (observer.current) {
+            observer.current.disconnect();
+        }
+        observer.current = new IntersectionObserver(([entry]) => {
+            if (entry.isIntersecting) {
+                setTocActive(entry.target.id);
+            }
+        }, options);
+        const { current: currentObserver } = observer;
+        targets.forEach((target) => currentObserver.observe(target));
+        return () => currentObserver.disconnect();
+    });
```

Comment:
Pass an empty array as the second argument of `useEffect`

Reply:
I was trying to do that and it doesn't work if we pass an empty array as the second argument. I read some articules of people using 'Intersection Observable API' in React Hook ([Articulo1](https://dev.to/producthackers/intersection-observer-using-react-49ko) and [Articulo2](https://non-traditional.dev/how-to-use-an-intersectionobserver-in-a-react-hook-9fb061ac6cb5)) and  they reference the element they are 'observing' at using `useRef` and pass it as the second argument. The elements that we want to observe are provided by the dotCMS API so I'm not sure what we have to pass as the second argument

Reply:
You need to do this:

```js
const options = React.useMemo(
    () => ({
        root: null,
        rootMargin: '0px 0px -70% 0px',
        threshold: 1
    }),
    []
);
useEffect(() => {
    const targets = document.querySelectorAll('h2,h3');
    if (observer.current) {
        observer.current.disconnect();
    }
    observer.current = new IntersectionObserver(([entry]) => {
        if (entry.isIntersecting) {
            setTocActive(entry.target.id);
        }
    }, options);
    const { current: currentObserver } = observer;
    targets.forEach((target) => currentObserver.observe(target));
    return () => currentObserver.disconnect();
    // eslint-disable-next-line react-hooks/exhaustive-deps
}, [observer.current, options]);
```

The linter will throw a warning about an 'unnecessary' dependency but this is actually necessary.